### PR TITLE
feat(interval): add searchable interval interface

### PR DESCRIPTION
**Pull request by adamsiwiec**
- Adds some quality-of-life improvements in the UI layout of the sidebar / profile controls.
- Adds ability to manually zoom using search for time sequence

### CODE REVIEW

File: src/timestamp.rs [hunk 1]
```diff
@@ -124,4 +142,26 @@ impl Interval {
     pub fn lerp(self, value: f32) -> Timestamp {
         Timestamp((value * (self.duration_ns() as f32)).round() as i64 + self.start.0)
     }
+
+    // convert a string like "500.0 s" to a timestamp
+    pub fn convert_str_to_timestamp(s: &str) -> Result<Timestamp, IntervalParseError> {
```
Comment (elliottslaughter) [inline src/timestamp.rs:147]: Maybe call this `parse_timestamp`?

File: src/timestamp.rs [hunk 2]
```diff
@@ -124,4 +142,26 @@ impl Interval {
...
+        let mut parts = s.split_whitespace();
+        let value = parts.next().ok_or(IntervalParseError::NoValue)?;
+        let value = value
+            .parse::<f32>()
```
Comment (elliottslaughter) [inline src/timestamp.rs:151]: Since it shouldn't cost us anything, I wonder if we should parse this as `f64`. It's unlikely, but theoretically possible that we could lose precision with `f32`. (Timestamps are 64-bit for a reason.)

File: src/timestamp.rs [hunk 3]
```diff
@@ -124,4 +142,26 @@ impl Interval {
...
+        let ns_per_ms = 1_000_000;
+        let ns_per_s = 1_000_000_000;
+        let ns = match unit.as_str() {
+            "ns" => value as i64,
+            "us" => (value * ns_per_us as f32) as i64,
+            "ms" => (value * ns_per_ms as f32) as i64,
+            "s" => (value * ns_per_s as f32) as i64,
+            _ => return Err(IntervalParseError::InvalidUnit),
+        };
+        Ok(Timestamp(ns))
```
Comment (elliottslaughter) [inline src/timestamp.rs:165]: Some general thoughts on the parser:

 * If I'm not mistaken, we don't ever check `parts` again, so something like `1.0 s asdf` will parse despite having garbage at the end.
 * It seems reasonable to write `1.0s` (no space) but currently that wouldn't parse.
 * Please write unit tests for this function, it is small, self-contained and (importantly) tricky enough we could easily miss a corner case.

File: src/app.rs [hunk 4]
```diff
@@ -815,13 +821,86 @@ impl Window {
...
+            .inner;
+
+        if start_res.lost_focus()
+            && cx.view_interval_start_buffer != cx.view_interval.start.to_string()
+        {
+            match Interval::convert_str_to_timestamp(&cx.view_interval_start_buffer) {
+                Ok(start) => {
+                    // validate timestamp
+                    if start > cx.view_interval.stop {
+                        cx.view_interval_start_buffer = "Start must be before stop".to_string();
```
Comment (elliottslaughter) [inline src/app.rs:847]: I was wondering if, instead of replacing the contents of the interval string, we should pop up an error message right below this widget. This has the advantage of not throwing away the user's entry if they typo'd something.

**Review by elliottslaughter — COMMENTED** [review]
Thanks for working on this! Initial comments in the code.
